paimon-core/src/main/java/org/apache/paimon/append/AppendOnlyWriter.java

```diff
@@ -215,6 +215,11 @@ public boolean isCompacting() {
         return compactManager.isCompacting();
     }
 
+    @Override
+    public boolean hasDelayedCompact() {
+        return compactManager.hasDelayedCompact();
```

Comment:
If Delayed Compact only apply pk table, then here shoule throw a exception?

Reply:
@wwj6591812 when clean unused writer, we need to call this method for conflict check.